src/FFMpeg/Media/Video.php

```diff
@@ -129,6 +129,56 @@ public function save(FormatInterface $format, $outputPathfile)
             }
         }
 
+        // Merge Filters into one command
+        $videoFilterVars = $videoFilterProcesses = [];
+        for($i=0;$i<count($commands);$i++) {
+            $command = $commands[$i];
+            if ( $command == '-vf' ) {
+                $commandSplits = explode(";", $commands[$i + 1]);
+                if ( count($commandSplits) == 1 ) {
+                    $commandSplit = $commandSplits[0];
+                    $command = trim($commandSplit);
+                    if ( preg_match("/^\[in\](.*?)\[out\]$/is", $command, $match) ) {
+                        $videoFilterProcesses[] = $match[1];
+                    } else {
+                        $videoFilterProcesses[] = $command;   
+                    }
+                } else {
+                    foreach($commandSplits as $commandSplit) {
+                        $command = trim($commandSplit);
+                        if ( preg_match("/^\[[^\]]+\](.*?)\[[^\]]+\]$/is", $command, $match) ) {
+                            $videoFilterProcesses[] = $match[1];
+                        } else {
+                            $videoFilterVars[] = $command;
+                        }                
+                    }
+                }
+                unset($commands[$i]);
+                unset($commands[$i + 1]);
+                $i++;
+            }
+        }
+        $videoFilterCommands = $videoFilterVars;
+        $lastInput = 'in';
+        foreach($videoFilterProcesses as $i => $process) {
+            $command = '[' . $lastInput .']';
+            $command .= $process;
+            $lastInput = 'p' . $i;
+            if ( $i == count($videoFilterProcesses) - 1 ) {
+                $command .= '[out]';
+            } else {
+                $command .= '[' . $lastInput . ']';
+            }
+            
+            $videoFilterCommands[] = $command;
+        }
+        $videoFilterCommand = implode(";", $videoFilterCommands);
+        
+        if ( $videoFilterCommand ) {
+            $commands[] = '-vf';
+            $commands[] = $videoFilterCommand;
+        }
```

Comment:
Hi @shtse8,

Thanks for this PR.

What are you trying to do here?
Why are you defining a command like this: `[in]scale=...[out]` to get rid of the [in] and [out] afterwards?
I think you should also wrap this in a test `if($commands !== NULL)` and use foreach instead of for.

To my opinion, this part is useless. The command generated by the filter should already be clean enough to be used by the save function, as is, without being processed in another way.

Can you paste here the desired command? Is it somethink like `ffmpeg -i input.avi -vf scale=320:240 output.avi`?
If yes, I think you don't need all this.